dask/array/tests/test_array_core.py

```diff
@@ -3366,6 +3366,18 @@ def test_zarr_roundtrip():
         assert a2.chunks == a.chunks
 
 
+def test_zarr_distributed_roundtrip():
+    zarr = pytest.importorskip('zarr')
+    distributed = pytest.importorskip('distributed')
+    with tmpdir() as d:
+        with distributed.Client(n_workers=2, threads_per_worker=1):
+            a = da.zeros((3, 3), chunks=(1, 1))
+            a.to_zarr(d)
+            a2 = da.from_zarr(d)
+            assert_eq(a, a2)
+            assert a2.chunks == a.chunks
```

Comment:
This should probably live in tests/test_distributed.py

Also, you should copy the models already in that file, or else read through http://distributed.readthedocs.io/en/latest/develop.html#writing-tests

When you start creating thousands of dask clusters and tornado IOLoops in a process weird things happen if you aren't careful.  The testing infrastructure in that doc page includes some well-trusted mechanisms.

Reply:
FWIW I found this usage pattern in that file. Not attached to the usage pattern or which file within Dask this lives. Just noting that usage pattern already exists in the code base.

Reply:
Sorry, was incorrect about the file. Found it here.

ref: https://github.com/dask/dask/blob/0.17.5/dask/bytes/tests/test_hdfs.py#L300-L312

Reply:
In any event, have moved this to Distributed and used the `with cluster()` syntax. Please let me know your thoughts on it.

Reply:
Ah, ok, thanks for hunting that down.

Anyway, I strongly suggest that we stick to the pattern in the dask.distributed documentation, or in the test_distributed.py file

Reply:
Sure, I agree. Only used it as it was the first thing I found and did not know what I was doing. :)

Reply:
Added a PR ( https://github.com/dask/dask/pull/3609 ) to try and refresh that test.